core/src/main/java/ai/timefold/solver/core/impl/heuristic/move/Move.java

```diff
@@ -31,11 +30,39 @@
  * An implementation must extend {@link AbstractMove} to ensure backwards compatibility in future versions.
  * It is highly recommended to override {@link #getPlanningEntities()} and {@link #getPlanningValues()},
  * otherwise the resulting move will throw an exception when used with Tabu search.
+ * <p>
+ * To ease interoperability with Neighborhoods API,
+ * this interface extends {@link ai.timefold.solver.core.preview.api.move.Move},
+ * giving the user an option to override certain methods which they should not.
+ * Specifically, the following methods must not be overridden by the user,
+ * as suitable default implementations are provided:
+ *
+ * <ul>
+ * <li>{@link #execute(MutableSolutionView)}</li>
+ * <li>{@link #rebase(Rebaser)}</li>
+ * <li>{@link #describe()}</li>
+ * </ul>
+ *
+ * The following methods should also not be overridden, on account of being deprecated:
+ *
+ * <ul>
+ * <li>{@link #doMove(ScoreDirector)}</li>
+ * </ul>
  *
+ * <p>
+ * This entire interface exists to provide interoperability with move selectors
+ * and will eventually be phased out in favor of the Neighborhoods API.
+ * It will be marked as deprecated for removal in a future release.
+ *
+ * <p>
+ * To avoid having to implement this interface and instead use the new Move API directly,
+ * you can use {@link MoveAdapters#toLegacyMoveIterator(Iterator)} in your move selectors.
+ * Note that {@link MoveAdapters} is not public API, just like this interface.
+ * 
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @see AbstractMove
  */
-public interface Move<Solution_> {
+public interface Move<Solution_> extends ai.timefold.solver.core.preview.api.move.Move<Solution_> {
```

Comment:
This is IMO the only interesting choice here.
Old moves automatically become new moves, reducing the need to adapt back and forth.

It makes this API a bit more complex, but IMO it's worth it. Also, this interface is not public API. (Which is crazy, considering that custom moves are an essential feature.)